build_compilers/test/build_test.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+@Tags(const ['presubmit-only'])
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+
+// TODO: Generalize this test and move it to package:build_test. This is copied
+// wholesale from the json_serializable package.
+void main() {
+  String pkgRoot;
+  try {
+    pkgRoot = _runProc('git', ['rev-parse', '--show-toplevel']);
+    var currentDir = Directory.current.resolveSymbolicLinksSync();
+    if (!p.isWithin(pkgRoot, currentDir)) {
+      throw new StateError('Expected the git root ($pkgRoot) '
+          'to be a parent of the current directory ($currentDir).');
+    }
+  } catch (e) {
+    print("Skipping this test – git didn't run correctly");
```

Comment:
[nit] don't mix quote types.

Reply:
This is just because its using a `'` inside the string, which is allowed

Reply:
There's a lint for that 😄

Reply:
oops, missed the `'`. Carry on, nothing to see here.